assets/src/js/listing_pages.js

```diff
@@ -17,4 +18,27 @@ export const setupListingPages = () => {
 
   listViewToggle.onclick = switchViews;
   gridViewToggle.onclick = switchViews;
+
+  // Setup filters for the News & Stories page.
+  const filters = document.querySelector('.listing-page-filters');
+  if (!filters) {
+    return;
+  }
+
+  // Functions and constants.
+  const CATEGORY_FILTER = 'category';
+
+  const updateFilter = event => {
+    const {target: {id, value}} = event;
+    const urlParams = new URLSearchParams(window.location.search);
+    if (value) {
+      urlParams.set(id, value);
+    } else {
+      urlParams.delete(id);
+    }
+    window.location.search = urlParams;
+  };
+
+  // Category filter.
+  document.getElementById(CATEGORY_FILTER).onchange = updateFilter;
```

Comment:
Should we check whether the element exist?

```suggestion
  const categoryElement = document.getElementById(CATEGORY_FILTER);
  if (categoryElement) {
    categoryElement.onchange = updateFilter;
  }
```

Reply:
Hm, we could, I didn't do that because [earlier](https://github.com/greenpeace/planet4-master-theme/pull/2422/files/015960f79578fba419a4e8fe8c71bc7d236eddf6#diff-dbed45bed73d24a7bc0816904ff6f7a067c5aad976a8b2baab355047778346dbR23) I check that the filters div is there